templates/etcd-ha/3/docker-compose.yml

```diff
@@ -1,41 +1,22 @@
 etcd:
-  image: rancher/etcd:v2.3.6
+  image: rancher/etcd:v2.3.6-4
   labels:
     io.rancher.scheduler.affinity:container_label_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
     io.rancher.sidekicks: data
-    io.rancher.service.allocate.skip.serialize: 'true'
-    io.rancher.container.start_once: 'true'
   environment:
     ETCD_DATA_DIR: /data
     ETCDCTL_ENDPOINT: http://etcd:2379
+    RANCHER_DEBUG: ${DEBUG}
   links:
   - data
-  - discovery
   volumes_from:
-    - data
-# containerize data volume to enable restarts and upgrades
+  - data
+
 data:
   image: busybox
-  command: /bin/true
+  entrypoint: /bin/true
```

Comment:
Combining entrypoint and command allows you to specify the default executable for your image while also providing default arguments to that executable which may be overridden by the user. Let's look at an example:

```
FROM ubuntu:trusty
ENTRYPOINT ["/bin/ping","-c","3"]
CMD ["localhost"]
```

In this case, busybox image has no default entrypoint or command. Therefore, using entrypoint vs command is a matter of stylistic choice. However, if busybox had a defined entrypoint that did something which might fail or have some other side effect, we would want to override that. Otherwise, as you may have experienced, the command will be sent as arguments to the entrypoint executable, which may or may not return 0. If it fails, we lose the data volume.

I chose to make the change in an attempt to standardize. If someone were to mimic our data volume settings with an image that had an entrypoint already defined, they could accidentally lose their data. The latest four k8s templates use it as well.